notebooks/ch12/LR_Case_Study.md

```diff
@@ -0,0 +1,792 @@
+
+## Linear Regression Model
+
+Since all variables seem to have a linear relationship with cirrhosis, we can just include all of them in our model. 
+
+Our model looks like this:
+
+$$
+f_\hat{\theta} (x) = \hat{\theta_0} + \hat{\theta_1} (Urban population) + \hat{\theta_2} (Late births) + \hat{\theta_3} (Wine) + \hat{\theta_3} (Liquor)
+$$
+
+Remember, the $\hat{\theta_0}$ is accounted for due to the bias column in our dataframe.
+
+Here are the functions we defined earlier in the multiple linear regression section, which we will use again:
+
+
+```python
+def linear_model(thetas, X):
+    '''Returns predictions by a linear model on x_vals.'''
+    return X @ thetas
+
+def mse_cost(thetas, X, y):
+    return np.mean((y - linear_model(thetas, X)) ** 2)
+
+def grad_mse_cost(thetas, X, y):
+    n = len(X)
+    return -2 / n * (X.T @ y  - X.T @ X @ thetas)
+```
+
+In order to use the above functions, we need `X`, and `y`. `thetas` is actually set to be an array of all 1's in the body of the `minimize` function. We can get these from our dataframe. Remember that `X` and `y` have to be numpy matrices in order to be able to multiply them with `@` notation.
+
+`X` consists of all columns of the dataframe except for the target column, `Cirrhosis`. Note that we also need to call `as_matrix()` so that `X` is a matrix.
+
+
+
+```python
+X = (drinking_data
+     .loc[:, ['Bias', 'Urban population', 'Late births', 'Wine', 'Liquor' ]]
+     .as_matrix())
+```
+
+`y` consists of the `Cirrhosis` column of our dataset.
+
+
+```python
+y = drinking_data['Cirrhosis'].as_matrix()
+```
+
+Now we just need to call the `minimize` function defined in a previous section.
+
+
+```python
+thetas = minimize(mse_cost, grad_mse_cost, X, y)
+```
+
+    theta: [ 0.  0.  0.  0.  0.] | cost: 4217.20
+    theta: [ 0.01  0.64  0.45  0.14  0.63] | cost: 999.01
+    theta: [ 0.    1.    0.39  0.59  0.21] | cost: 955.60
+    theta: [-0.02  0.86  0.1   1.58  0.34] | cost: 902.97
+    theta: [-0.01  0.41  0.82  2.13  0.13] | cost: 809.45
+    theta: [-0.08  0.55  0.54  1.86  0.19] | cost: 634.59
+    theta: [-0.33  0.43  0.63  1.86  0.11] | cost: 339.56
+    theta: [-1.26  0.22  0.76  1.8  -0.01] | cost: 98.33
+    theta: [-3.18  0.2   0.85  1.81 -0.02] | cost: 97.67
+    theta: [-7.49  0.17  1.01  1.81 -0.04] | cost: 96.69
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    
+
+Looks like gradient descent converged to those theta values! Thus, our linear model is:
+
+$y = -11.19 + 0.14x_1 + 1.16x_2 + 1.81x_3 - 0.05x_4$
+
+Let's compare this equation that we obtained to the one we would get if we had used `sklearn`'s LinearRegression model instead.
+
+
+```python
+model = LinearRegression()
+model.fit(X, y)
+model.coef_
+```
+
+
+
+
+    array([ 0.  ,  0.14,  1.16,  1.81, -0.05])
```

Comment:
how would you explain the 0 vs -11.19?

Reply:
i honestly have no idea why this happened >_>

Reply:
help

Reply:
@ashleychien @ashleychien1 sorry I'm dumb... LOL i didn't press submit review lol